static/app/components/modals/sentryAppPublishRequestModal.tsx

```diff
@@ -59,11 +59,11 @@ type Props = ModalRenderProps & {
   app: SentryApp;
 };
 
-export default class SentryAppPublishRequestModal extends Component<Props> {
-  form = new FormModel({transformData});
+export default function SentryAppPublishRequestModal(props: Props) {
+  const [form] = useState<FormModel>(new FormModel({transformData}));
```

Comment:
you want to use an initializer on these, even though it doesn't update state on rerender it creates a new FormModel every render and doesn't use it.
```suggestion
  const [form] = useState<FormModel>(() => new FormModel({transformData}));
```